### PR TITLE
Add file status for API "Get a single commit from a repository" (#16205)

### PR DESCRIPTION
**Pull request by jeremiepozzigithub**
#16205 To obtain a closer behavior to the api from github, the status (added, modified, removed) of a file should be available in addition to the filename.
See github doc :
https://docs.github.com/fr/rest/commits/commits?apiVersion=2022-11-28#get-a-commit

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: services/convert/git_commit.go [hunk 1]
```diff
@@ -198,8 +207,17 @@ func ToCommit(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Rep
 		affectedFileList := make([]*api.CommitAffectedFiles, 0, len(fileStatus.Added)+len(fileStatus.Removed)+len(fileStatus.Modified))
 		for _, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} {
 			for _, filename := range files {
+				var filestatus = "unknown"
+				if contains(fileStatus.Added, filename) {
```
Comment (lunny) [inline services/convert/git_commit.go]: You should use `util.SliceContainsString` directly.

Comment (jeremiepozzigithub) [reply services/convert/git_commit.go]: Thanks. I updated the code accordingly

File: services/convert/git_commit.go [hunk 2]
```diff
@@ -198,8 +198,17 @@ func ToCommit(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Rep
...
 		for _, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} {
 			for _, filename := range files {
+				var filestatus = "unknown"
+				if util.SliceContainsString(fileStatus.Added, filename) {
+					filestatus = "added"
+				} else if util.SliceContainsString(fileStatus.Removed, filename) {
+					filestatus = "removed"
+				} else if util.SliceContainsString(fileStatus.Modified, filename) {
+					filestatus = "modified"
+				}
```
Comment (wxiaoguang) [inline services/convert/git_commit.go]: No need to use "SliceContainsString"

In the outer loop `for _, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} `, you already know current file is either added, removed or modified.

Comment (jeremiepozzigithub) [reply services/convert/git_commit.go]: Ok I modifed the code accordingly.

File: services/convert/git_commit.go [hunk 3]
```diff
@@ -196,10 +196,22 @@ func ToCommit(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Rep
 		}
 
 		affectedFileList := make([]*api.CommitAffectedFiles, 0, len(fileStatus.Added)+len(fileStatus.Removed)+len(fileStatus.Modified))
-		for _, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} {
+		for sliceIndex, files := range [][]string{fileStatus.Added, fileStatus.Removed, fileStatus.Modified} {
```
Comment (wxiaoguang) [inline services/convert/git_commit.go]:
```suggestion
		for filestatus, files := range map[string][]string{"added": fileStatus.Added, "removed": fileStatus.Removed, "modified": fileStatus.Modified} {
```

File: services/convert/git_commit.go [hunk 4]
```diff
@@ -196,10 +196,22 @@ func ToCommit(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Rep
...
+				switch sliceIndex {
+					case 0:
+						filestatus = "added"
+					case 1:
+						filestatus = "removed"
+					case 2:
+						filestatus = "modified"
+					default:
+						filestatus = "unknown"
+				}
```
Comment (wxiaoguang) [inline services/convert/git_commit.go]:
```suggestion
```

**Review by wxiaoguang — APPROVED** [review]

**Comment by lunny** [comment]
Please fix the CI with `make fmt`

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**